File: java/Valid Parentheses.py
class Solution:
    # @return a boolean
    def isValid(self, s):
        count = 0;
        small = 0;
        mid = 0;
        big = 0;
        tmp = [];
        
        while count < len(s):
            if s[count] == '(':
                #small = small + 1;
                tmp.append(1);
            elif s[count] == ')':
                #small = small - 1;
                if len(tmp) == 0:
                    return False;
                if tmp.pop() != 1:
                    return False;
                
            elif s[count] == '[':
                #mid = mid + 1;
                tmp.append(2);
            elif s[count] == ']':
                #mid = mid - 1;
                if len(tmp) == 0:
                    return False;
                if tmp.pop() != 2:
                    return False;
                
            elif s[count] == '{':
                #big = big + 1;
                tmp.append(3);
            elif s[count] == '}':
                #big = big - 1;
                if len(tmp) == 0:
                    return False;
                if tmp.pop() != 3:
                    return False;
            count = count + 1;  
                
        if len(tmp) == 0:
            return True;
        else:
            return False;
                
                

File: cpp/char_problem/validParentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        int small = 0,mid = 0,big = 0;
        stack<char> st;
        
        for(int i = 0;i < s.length();i++)
        {
            if(s[i] == ')')
            {
                if(st.size() != 0 && st.top() == '(')
                    st.pop();
                else
                    return false;
            }
            else if(s[i] == ']')
            {
                if(st.size() != 0 && st.top() == '[')
                    st.pop();
                else
                    return false;
            }
            else if(s[i] == '}')
            {
                if(st.size() != 0 && st.top() == '{')
                    st.pop();
                else
                    return false;
            }   
            else
                st.push(s[i]);
        }
        
        return st.size() == 0;
        
    }
};